components/timekit/sources/booking-with-state/booking-with-state.mjs

```diff
@@ -0,0 +1,97 @@
+import timekit from "../../timekit.app.mjs";
+import crypto from "crypto";
+
+const docLink = "https://developers.timekit.io/reference/graphs";
+
+export default {
+  key: "timekit-booking-with-state",
+  name: "New Booking with State",
+  description: `Emit new event when a booking has a specific state. [See the docs](${docLink}).`,
+  version: "0.0.1",
+  type: "source",
+  props: {
+    timekit,
+    db: "$.service.db",
+    http: "$.interface.http",
+    graph: {
+      propDefinition: [
+        timekit,
+        "graph",
+      ],
+    },
+    state: {
+      propDefinition: [
+        timekit,
+        "state",
+        (c) => ({
+          graph: c.graph,
+        }),
+      ],
+    },
+    secret: {
+      type: "string",
+      label: "Secret",
+      description: "Configured secret to verify webhook events. If left blank, no verification will be done.",
```

Comment:
I'd suggest adding some information here about where to find the `secret`. I copied the info below from here - https://developers.timekit.io/docs/signed-webhooks
```suggestion
      description: "Configured secret to verify webhook events. If left blank, no verification will be done. Navigate to API Settings > Keys and look for the last card on the page. The secret shown there is the key you'll use",
```